addons/rose/addon/styles/hds/overrides.scss

```
@@ -16,3 +16,10 @@
     }
   }
 }
+
+// removes field bottom margin (for tables with checkbox and badge in a cell)
+.hds-table__td {
+  .hds-form-field--layout-flag {
```

This suggestion overrides the exact selector used to add margin in the first place, scoped by table.
```suggestion
  [class^=hds-form-field], [class^=hds-form-group] {
```